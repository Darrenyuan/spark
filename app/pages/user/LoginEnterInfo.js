import React from 'react';

import {
  StyleSheet,
  Text,
  View,
  Image,
  ImageBackground,
  TextInput,
  TouchableOpacity,
  Keyboard,
  Platform,
} from 'react-native';
import { Avatar, Icon } from 'react-native-elements';
import md5 from 'react-native-md5';
import ImageCropPicker from 'react-native-image-crop-picker';
import styleUtil from '../../common/styleUtil';
import NavigatorPage from '../../components/NavigatorPage';
import OverlayModal from '../../components/OverlayModal';
import config from '../../common/config';
import navigate from '../../screens/navigate';
import DatePicker from '../../components/DatePicker';
import LoginMoreInfo from './LoginMoreInfo';
import { apiEditRegistInfo1 } from '../../services/axios/api';

export default class LoginEnterInfo extends NavigatorPage {
  static defaultProps = {
    ...NavigatorPage.navigatorStyle,
    navigationBarInsets: false,
    style: { backgroundColor: 'transparent', borderBottomWidth: 0 },
    scene: navigate.sceneConfig.PushFromRight,
    leftView: (
      <TouchableOpacity
        style={{ paddingLeft: 10 }}
        onPress={_ => {
          navigate.pop();
        }}
      >
        <Icon name="ios-arrow-back" type="ionicon" color="white" size={25} />
      </TouchableOpacity>
    ),
  };

  constructor(props) {
    super(props);
    this.state = {
      face: null,
      nickName: '',
      birth: '',
      sex: '男',
    };
  }

  // TODO
  _netRegisterInfo1 = () => {
    const { face, nickName, birth, sex } = this.state;
    console.log(face);
    const auid = '';
    const M9 = new Date().getTime();
    const strM9 = `${M9}`;
    const option = {
      nickName,
      birth,
      sex,
      face: {},
      auid,
      M0: Platform.OS === 'ios' ? 'IMMC' : 'MMC',
      M2: '',
      M3: '120.45435,132.32424',
      M8: md5.hex_md5(auid + strM9),
      M9: strM9,
    };
    if (face !== null) {
      option.face.uri = face.path;
    } else {
      option.face.uri = '../../assets/image/login_default_avatar.png';
    }
    toast.modalLoading();
    console.log(option);

    apiEditRegistInfo1(option).then(res => {
      toast.modalLoadingHide();
      console.log(res.data);
      console.log('qqqqqqqqqqq');
      if (res.data.code === 1) {
        // TODO  同步redux
        // config.setStatusAndMarker(res.data);
        navigate.pushNotNavBar(LoginMoreInfo, {
          phone: this.props.phone,
        });
      }
    });
  };

  _btnStyle = bool => (bool ? styleUtil.themeColor : styleUtil.disabledColor);

  _openCamera = type => {
    ImageCropPicker.openCamera({
      cropping: true,
      // compressImageQuality: 1
    }).then(image => {
      this.setState({ face: image });
    });
  };

  _selectLibrary = type => {
    ImageCropPicker.openPicker({
      multiple: false,
      // cropping: true,
      mediaType: 'photo',
      compressImageQuality: Platform.OS === 'ios' ? 0 : 1,
      minFiles: 1,
      maxFiles: 1,
    })
      .then(image => {
        this.setState({ face: image });
      })
      .catch(err => {
        if (err.code === 'E_PICKER_CANCELLED') {
        }
      });
  };

  _onClickAvatar = (type = 'avatar') => {
    const items = [
      {
        title: '拍照',
        onPress: _ => config.loadData(_ => this._openCamera(type)),
      },
      {
        title: '从相册中选取',
        onPress: _ => config.loadData(_ => this._selectLibrary(type)),
      },
    ];
    config.showAction(items);
  };

  showDatePicker = () => {
    let { birth } = this.state;
    const arr = birth.split('/');
    OverlayModal.show(
      <DatePicker
        selectedYear={arr[0]}
        selectedMonth={arr[1]}
        selectedDate={arr[2]}
        onDone={arr => {
          birth = arr.join('/');
          this.setState({ birth });
        }}
      />,
    );
  };

  renderPage() {
    const { face, nickName, birth, sex } = this.state;

    return (
      <TouchableOpacity
        style={styleUtil.container}
        activeOpacity={1}
        onPress={_ => {
          Keyboard.dismiss();
        }}
      >
        <View style={{ overflow: 'hidden' }}>
          <ImageBackground
            style={{
              width: styleUtil.window.width,
              height: styleUtil.window.width * (222.0 / 375.0),
              resizeMode: 'contain',
              justifyContent: 'flex-end',
              alignItems: 'center',
            }}
            source={require('../../assets/image/login_head_background.png')}
          >
            <TouchableOpacity
              style={{
                flexDirection: 'column',
                alignItems: 'center',
                marginBottom: 20,
              }}
              onPress={_ => {
                this._onClickAvatar();
                Keyboard.dismiss();
              }}
            >
              <Avatar size={71} rounded source={config.defaultAvatar(face ? face.path : '')} />
              <Text style={{ fontSize: 14, color: 'white', marginTop: 10 }}>修改头像</Text>
            </TouchableOpacity>
          </ImageBackground>

          <View style={{ marginTop: 80, marginHorizontal: 40 }}>
            <View
              style={{
                flexDirection: 'row',
                alignItems: 'center',
                justifyContent: 'flex-end',
              }}
            >
              <Image source={require('../../assets/image/login_nick_name.png')} />
              <TextInput
                placeholder="请输入你的昵称"
                placeholderTextColor="#E5E5E5"
                autoCorrect={false}
                underlineColorAndroid="transparent"
                style={[styles.inputField, { flex: 1 }]}
                value={nickName}
                maxLength={11}
                onChangeText={text => {
                  this.setState({ nickName: text });
                }}
              />
            </View>
            <View
              style={{
                flexDirection: 'row',
                alignItems: 'center',
                marginTop: 40,
                justifyContent: 'flex-end',
              }}
            >
              <Image source={require('../../assets/image/login_birthday.png')} />

              <TextInput
                editable={false}
                placeholder="请选择你的生日"
                placeholderTextColor="#E5E5E5"
                autoCorrect={false}
                underlineColorAndroid="transparent"
                style={[styles.inputField, { flex: 1 }]}
                value={birth}
                maxLength={11}
              />
              <TouchableOpacity
                style={{
                  width: '100%',
                  height: '100%',
                  position: 'absolute',
                }}
                onPress={_ => {
                  this.showDatePicker();
                  Keyboard.dismiss();
                }}
              />
            </View>
            <View
              style={{
                flexDirection: 'row',
                marginTop: 30,
                justifyContent: 'center',
                alignItems: 'center',
              }}
            >
              <TouchableOpacity
                style={{ flexDirection: 'row', alignItems: 'center' }}
                onPress={_ => {
                  this.setState({ sex: '男' });
                  Keyboard.dismiss();
                }}
              >
                <Icon
                  name={sex == '男' ? 'ios-radio-button-on' : 'ios-radio-button-off'}
                  type="ionicon"
                  color={styleUtil.themeColor}
                  size={25}
                />
                <Text style={{ fontSize: 16, color: '#454545', marginLeft: 15 }}>男性</Text>
              </TouchableOpacity>
              <TouchableOpacity
                style={{
                  flexDirection: 'row',
                  alignItems: 'center',
                  marginLeft: 60,
                }}
                onPress={_ => {
                  this.setState({ sex: '女' });
                  Keyboard.dismiss();
                }}
              >
                <Icon
                  name={sex == '女' ? 'ios-radio-button-on' : 'ios-radio-button-off'}
                  type="ionicon"
                  color={styleUtil.themeColor}
                  size={25}
                />
                <Text style={{ fontSize: 16, color: '#454545', marginLeft: 15 }}>女性</Text>
              </TouchableOpacity>
            </View>
            <TouchableOpacity
              activeOpacity={nickName.length > 0 && birth.length > 0 ? 0.5 : 1}
              style={[
                styles.buttonBox,
                {
                  backgroundColor: this._btnStyle(nickName.length > 0 && birth.length > 0),
                  borderColor: this._btnStyle(nickName.length > 0 && birth.length > 0),
                },
              ]}
              onPress={_ => {
                if (nickName.length > 0 && birth.length > 0) {
                  this._netRegisterInfo1();
                }
              }}
            >
              <Text style={styles.buttonText}>下一步</Text>
            </TouchableOpacity>
          </View>
        </View>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: styleUtil.backgroundColor,
  },
  signUpBox: {
    marginTop: 10,
    // padding: 10
  },
  title: {
    marginBottom: 20,
    color: '#333',
    fontSize: 20,
    textAlign: 'center',
  },
  inputField: {
    marginLeft: 8,
    height: 44,
    paddingLeft: 8,
    color: '#454545',
    fontSize: 16,
    backgroundColor: 'transparent',
    borderBottomWidth: styleUtil.borderSeparator,
    // borderWidth: styleUtil.borderSeparator,
    borderColor: styleUtil.borderColor,
  },
  buttonBox: {
    marginTop: 80,
    backgroundColor: styleUtil.themeColor,
    height: 48,
    borderWidth: 1,
    borderColor: styleUtil.themeColor,
    borderRadius: 24,
  },
  buttonText: {
    fontSize: 20,
    color: '#fff',
    textAlign: 'center',
    marginTop: 12,
  },
  verifyCodeBox: {
    flexDirection: 'row',
    marginTop: 10,
    justifyContent: 'space-between',
  },
  countBtn: {
    width: 110,
    height: 40,
    padding: 10,
    marginLeft: 8,
    borderWidth: 1,
    borderColor: styleUtil.themeColor,
    backgroundColor: styleUtil.themeColor,
    borderRadius: 4,
  },
  countBtnText: {
    textAlign: 'center',
    color: '#fff',
    fontSize: 16,
  },
  closeModal: {
    position: 'absolute',
    bottom: 20,
    alignSelf: 'center',
  },
});
