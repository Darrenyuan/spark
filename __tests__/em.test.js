import React from 'react';
import LoginEnterInfo from '../app/pages/user/LoginEnterInfo';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('Testing LoginEnterInfo component', () => {
  it('renders as expected', () => {
    console.log('enter render');
    const tree = renderer.create(<LoginEnterInfo />).toJSON();
    expect(tree).toMatchSnapshot();
    console.log('after render');
    // expect(4).toBeTruthy();
  });
});
