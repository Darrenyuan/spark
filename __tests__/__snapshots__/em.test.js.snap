// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LoginEnterInfo component renders as expected 1`] = `
<View
  leftView={
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "paddingLeft": 10,
        }
      }
    >
      <Themed.Icon
        color="white"
        name="ios-arrow-back"
        size={25}
        type="ionicon"
      />
    </TouchableOpacity>
  }
  navBarHidden={false}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#f8f8f8",
        "flex": 1,
      },
      Object {
        "backgroundColor": "transparent",
        "borderBottomWidth": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "marginTop": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "alignItems": "center",
                "height": 444,
                "justifyContent": "flex-end",
                "resizeMode": "contain",
                "width": 750,
              }
            }
          >
            <Image
              source="test-file-stub"
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": 444,
                    "width": 750,
                  },
                  undefined,
                ]
              }
            />
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "column",
                  "marginBottom": 20,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 35.5,
                    "height": 71,
                    "width": 71,
                  }
                }
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#bdbdbd",
                      "borderRadius": 35.5,
                      "flex": 1,
                      "overflow": "hidden",
                      "position": "relative",
                    }
                  }
                >
                  <Image
                    onLoadEnd={[Function]}
                    source="test-file-stub"
                    style={
                      Object {
                        "flex": 1,
                        "height": null,
                        "width": null,
                      }
                    }
                    theme={
                      Object {
                        "colors": Object {
                          "disabled": "hsl(208, 8%, 90%)",
                          "divider": "#bcbbc1",
                          "error": "#ff190c",
                          "grey0": "#393e42",
                          "grey1": "#43484d",
                          "grey2": "#5e6977",
                          "grey3": "#86939e",
                          "grey4": "#bdc6cf",
                          "grey5": "#e1e8ee",
                          "greyOutline": "#bbb",
                          "platform": Object {
                            "android": Object {
                              "error": "#f44336",
                              "primary": "#2196f3",
                              "secondary": "#9C27B0",
                              "success": "#4caf50",
                              "warning": "#ffeb3b",
                            },
                            "ios": Object {
                              "error": "#ff3b30",
                              "primary": "#007aff",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                          },
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#8F0CE8",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      }
                    }
                  />
                  <View
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#bdbdbd",
                          "flex": 1,
                          "height": null,
                          "justifyContent": "center",
                          "width": null,
                        }
                      }
                      testID="RNE__Image__placeholder"
                    />
                  </View>
                </View>
              </View>
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 14,
                    "marginTop": 10,
                  }
                }
              >
                修改头像
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "marginHorizontal": 40,
                "marginTop": 80,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }
            >
              <Image
                source="test-file-stub"
              />
              <TextInput
                allowFontScaling={true}
                autoCorrect={false}
                maxLength={11}
                onChangeText={[Function]}
                placeholder="请输入你的昵称"
                placeholderTextColor="#E5E5E5"
                rejectResponderTermination={true}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 0.5,
                      "borderColor": "#CCC",
                      "color": "#454545",
                      "fontSize": 16,
                      "height": 44,
                      "marginLeft": 8,
                      "paddingLeft": 8,
                    },
                    Object {
                      "flex": 1,
                    },
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "marginTop": 40,
                }
              }
            >
              <Image
                source="test-file-stub"
              />
              <TextInput
                allowFontScaling={true}
                autoCorrect={false}
                editable={false}
                maxLength={11}
                placeholder="请选择你的生日"
                placeholderTextColor="#E5E5E5"
                rejectResponderTermination={true}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 0.5,
                      "borderColor": "#CCC",
                      "color": "#454545",
                      "fontSize": 16,
                      "height": 44,
                      "marginLeft": 8,
                      "paddingLeft": 8,
                    },
                    Object {
                      "flex": 1,
                    },
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "height": "100%",
                    "opacity": 1,
                    "position": "absolute",
                    "width": "100%",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginTop": 30,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "opacity": 1,
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                    theme={
                      Object {
                        "colors": Object {
                          "disabled": "hsl(208, 8%, 90%)",
                          "divider": "#bcbbc1",
                          "error": "#ff190c",
                          "grey0": "#393e42",
                          "grey1": "#43484d",
                          "grey2": "#5e6977",
                          "grey3": "#86939e",
                          "grey4": "#bdc6cf",
                          "grey5": "#e1e8ee",
                          "greyOutline": "#bbb",
                          "platform": Object {
                            "android": Object {
                              "error": "#f44336",
                              "primary": "#2196f3",
                              "secondary": "#9C27B0",
                              "success": "#4caf50",
                              "warning": "#ffeb3b",
                            },
                            "ios": Object {
                              "error": "#ff3b30",
                              "primary": "#007aff",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                          },
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#8F0CE8",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      }
                    }
                    underlayColor="white"
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#FE7733",
                            "fontSize": 25,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "fontFamily": "Ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="iconIcon"
                    >
                      
                    </Text>
                  </View>
                </View>
                <Text
                  style={
                    Object {
                      "color": "#454545",
                      "fontSize": 16,
                      "marginLeft": 15,
                    }
                  }
                >
                  男性
                </Text>
              </View>
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "marginLeft": 60,
                    "opacity": 1,
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                    theme={
                      Object {
                        "colors": Object {
                          "disabled": "hsl(208, 8%, 90%)",
                          "divider": "#bcbbc1",
                          "error": "#ff190c",
                          "grey0": "#393e42",
                          "grey1": "#43484d",
                          "grey2": "#5e6977",
                          "grey3": "#86939e",
                          "grey4": "#bdc6cf",
                          "grey5": "#e1e8ee",
                          "greyOutline": "#bbb",
                          "platform": Object {
                            "android": Object {
                              "error": "#f44336",
                              "primary": "#2196f3",
                              "secondary": "#9C27B0",
                              "success": "#4caf50",
                              "warning": "#ffeb3b",
                            },
                            "ios": Object {
                              "error": "#ff3b30",
                              "primary": "#007aff",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                          },
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#8F0CE8",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      }
                    }
                    underlayColor="white"
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#FE7733",
                            "fontSize": 25,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "fontFamily": "Ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="iconIcon"
                    >
                      
                    </Text>
                  </View>
                </View>
                <Text
                  style={
                    Object {
                      "color": "#454545",
                      "fontSize": 16,
                      "marginLeft": 15,
                    }
                  }
                >
                  女性
                </Text>
              </View>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#BBB",
                  "borderColor": "#BBB",
                  "borderRadius": 24,
                  "borderWidth": 1,
                  "height": 48,
                  "marginTop": 80,
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 20,
                    "marginTop": 12,
                    "textAlign": "center",
                  }
                }
              >
                下一步
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomColor": "#337ab7",
          "borderBottomWidth": 0,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "space-between",
          "left": 0,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 20,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "left": 0,
            "opacity": 1,
            "paddingLeft": 4,
            "paddingRight": 4,
            "position": "absolute",
            "right": 0,
            "top": 20,
          }
        }
      />
      <View
        onLayout={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "paddingLeft": 10,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
              underlayColor="white"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 25,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "bottom": 0,
          "left": 0,
          "right": 0,
        },
        Object {
          "height": 0,
        },
      ]
    }
  />
</View>
`;
